### PR TITLE
Add kafka log driver

### PR DESCRIPTION
**Pull request by gao-feng**
Signed-off-by: Gao feng <omarapazanadi@gmail.com>

### CODE REVIEW

File: kafkalog/kafka.go [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+	hosts     string
+	mu        sync.Mutex
+	readers   map[*logger.LogWatcher]struct{} // stores the active log followers
+}
+
+func (k *KafkaLogger) Log(msg *logger.Message) error {
+	select {
+	case k.logs <- msg:
+	default:
+		return fmt.Errorf("log chan is full")
```
Comment (gao-feng) [inline kafkalog/kafka.go]: log chan can be full even the size of channel is 1024. 
I think we should wait until logs are wrote to log chan, slow down tty output doesn't harm too much.

Comment (gnawux) [reply kafkalog/kafka.go]: Does this mean Kafka connection is slow? Do we need a connection pool?

Comment (gnawux) [reply kafkalog/kafka.go]: @gao-feng OK, I agree block more. Then

if we hold the log here, we need to decide when should we drop the message, and we shouldn't hold it if the kafka is down. Will the consumer drain the `chan` if the connection is broken?

Other options:

- use longer queue, such as 4096 or longer. But I hate async log.
- use a kafka client pool, multiple clients write logs for different container, however, this won't help for a single crazy logging container.

Comment (gao-feng) [reply kafkalog/kafka.go]: @gnawux thttps://github.com/gao-feng/hypercontainer-utils/blob/4432fa2dcfedd969410c75da425f6d07a6ced41c/kafkalog/kafka.go#L146 log will be dropped if kafka is down.

**Comment by gao-feng** [comment]
test result

jsonfile
https://gist.github.com/gao-feng/3847dd6358fb603c3029e981b9584b0b

kafka
https://gist.github.com/gao-feng/b38ba49647650311c562730076a2a221

**Comment by gnawux** [comment]
merge it

**Pull request merged**